node/core/pvf/common/src/worker/mod.rs

```diff
@@ -317,4 +317,92 @@ pub mod thread {
 			Some(*result.0)
 		}
 	}
+
+	#[cfg(test)]
+	mod tests {
+		use super::*;
+		use crate::worker::cpu_time_monitor_loop;
+		use cpu_time::ProcessTime;
+		use std::sync::mpsc::channel;
+
+		#[test]
+		fn get_condvar_should_be_pending() {
+			let condvar = get_condvar();
+			let outcome = *condvar.0.lock().unwrap();
+			assert!(outcome.is_pending());
+		}
+
+		#[test]
+		fn wait_for_threads_with_timeout_return_none_on_time_out() {
+			let condvar = Arc::new((Mutex::new(WaitOutcome::Pending), Condvar::new()));
+			let outcome = wait_for_threads_with_timeout(&condvar, Duration::from_secs(1));
+			assert!(matches!(outcome, None));
+		}
+
+		#[test]
+		fn wait_for_threads_with_timeout_returns_outcome() {
+			let condvar = Arc::new((Mutex::new(WaitOutcome::Pending), Condvar::new()));
+			let condvar2 = Arc::clone(&condvar);
+			thread::spawn(move || {
+				cond_notify_all(Arc::clone(&condvar2), WaitOutcome::Finished);
+			});
+			let outcome = wait_for_threads_with_timeout(&condvar, Duration::from_secs(2));
+			assert!(matches!(outcome.unwrap(), WaitOutcome::Finished));
+		}
+
+		#[test]
+		fn spawn_worker_thread_should_notify_on_done() {
+			let condvar = Arc::new((Mutex::new(WaitOutcome::Pending), Condvar::new()));
+			let response = spawn_worker_thread(
+				"thread",
+				move || 2,
+				Arc::clone(&condvar),
+				WaitOutcome::TimedOut,
+			);
+			let (lock, _) = &*condvar;
+			let r = response.unwrap().join().unwrap();
+			assert!(matches!(r, 2));
+			assert!(matches!(*lock.lock().unwrap(), WaitOutcome::TimedOut));
+		}
+
+		#[test]
+		fn spawn_worker_thread_should_not_notify() {
+			let condvar = Arc::new((Mutex::new(WaitOutcome::Finished), Condvar::new()));
+			let response = spawn_worker_thread(
+				"thread",
+				move || 2,
+				Arc::clone(&condvar),
+				WaitOutcome::TimedOut,
+			);
+			let (lock, _) = &*condvar;
+			let r = response.unwrap().join().unwrap();
+			assert!(matches!(r, 2));
```

Comment:
Nit: In your every test, `assert!(matches!(val, variant))` can be changed to `assert_eq!(val, variant)` (as there are no parametric enum variants involved). It's not a big deal for tests, but in production code, `matches!()` could potentially produce more overhead than a simple comparison. Also, using `matches!()` with a primitive value, like in this case, is quite unusual way to compare for equivalence.